tests/functional/aws-node-sdk/test/object/corsHeaders.js

```diff
@@ -0,0 +1,756 @@
+                    Origin: allowedOrigin,
+                    Test: 'test',
+                    Expires: 86400,
+                };
+                const headersResponse = {
+                    'access-control-allow-origin': allowedOrigin,
+                    'access-control-allow-methods': 'GET',
+                    'access-control-allow-credentials': 'true',
+                    vary,
+                };
+                methodRequest({ method: 'GET', bucket, headers, headersResponse,
+                code: 200 }, done);
+            });
+        });
+
+        describe('cors configuration : MaxAgeSeconds', () => {
+            const corsParams = {
+                Bucket: bucket,
+                CORSConfiguration: {
+                    CORSRules: [
+                        {
+                            AllowedMethods: [
+                                'GET',
+                            ],
+                            AllowedOrigins: [
+                                allowedOrigin,
+                            ],
+                            MaxAgeSeconds: 86400,
+                        },
+                    ],
+                },
+            };
+            beforeEach(done => {
+                s3.putBucketCors(corsParams, done);
+            });
+
+            afterEach(done => {
+                s3.deleteBucketCors({ Bucket: bucket }, done);
+            });
+
+            it('should respond with access-control-max-age header ' +
+            'if request matches CORS rule', done => {
+                const headers = { Origin: allowedOrigin };
+                const headersResponse = {
+                    'access-control-allow-origin': allowedOrigin,
+                    'access-control-allow-methods': 'GET',
+                    'access-control-allow-credentials': 'true',
+                    'access-control-max-age': '86400',
+                    vary,
+                };
+                methodRequest({ method: 'GET', bucket, headers, headersResponse,
+                code: 200 }, done);
+            });
+        });
+
+        describe('cors configuration : ExposeHeaders', () => {
+            const corsParams = {
+                Bucket: bucket,
+                CORSConfiguration: {
+                    CORSRules: [
+                        {
+                            AllowedMethods: [
+                                'GET',
+                            ],
+                            AllowedOrigins: [
+                                allowedOrigin,
+                            ],
+                            ExposeHeaders: [
+                                'Content-Type',
+                            ],
+                        },
+                    ],
+                },
+            };
+            beforeEach(done => {
+                s3.putBucketCors(corsParams, done);
+            });
+
+            afterEach(done => {
+                s3.deleteBucketCors({ Bucket: bucket }, done);
+            });
```

Comment:
I would place the `beforeEach` in the parent `describe` but the cors configuration (`corsParams`) being put in the `putBucketCors` changes with each test group. I can at least put the `afterEach` in the parent `describe`.

Another possibility is doing a `forEach` statement. I'm not sure how readable it would be to make one that would work for the whole group but the last two tests are pretty similar.